src/v/s3/client.cc

```diff
@@ -744,12 +754,14 @@ ss::future<client_pool::client_lease> client_pool::acquire() {
     vassert(!_pool.empty(), "'acquire' invariant is broken");
     auto client = _pool.back();
     _pool.pop_back();
+    auto client_it = _leased.insert(_leased.end(), client);
     co_return client_lease{
       .client = client,
       .deleter = ss::make_deleter(
-        [pool = weak_from_this(), client, g = std::move(guard)] {
+        [pool = weak_from_this(), client, client_it, g = std::move(guard)] {
             if (pool) {
```

Comment:
How does safe shutdown sequence look like? I see calls like

```
auto [client, deleter] = co_await _pool.acquire();
```

in `remote` and it seems hard to track all of the deleters on the stack. i guess the case to worry about would be if std::list were destroyed (via client_pool destructor) before the deleters were destroyed.

Reply:
I replaced list with `intrusive_list`